src/lib/pmd/CustomRuleRegistrar.ts

```diff
@@ -0,0 +1,83 @@
+import fs = require('fs');
+import path = require('path');
+import util = require('util');
+
+export const CUSTOM_RULE_REGISTER = path.join('catalogs', '.custom_paths.json');
+
+/**
+ * Handles registering custom rules into CustomRuleRegister JSON
+ * 
+ * TODO: verify validity of path. Better to fail now than later while creating catalog.
+ */
+
+export class CustomRuleRegistrar {
+
+    public async createEntries(language: string, location: string[]) {
+
+        // Fetch current entries in custom rule register as a Map
+        var currentEntries = await this.readCurrentEntries();
+
+        // If current language has entries, append new paths to existing entry
+        if (currentEntries.has(language)) {
+            var currentValue = currentEntries.get(language);
+            location.forEach((item) => {
+                currentValue.push(item);
+            });
+
+        } else {
+            // When current language does not exist, create a new entry
+            currentEntries.set(language, location);
+        }
+        
+        // Write updated Map to file
+        await this.writeToJson(this.mapToJson(currentEntries));
+
+    }
+
+    async readCurrentEntries(): Promise<Map<string, string[]>> {
+        var mapFromFile = new Map<string, string[]>();
+
+        const readFilePromise = util.promisify(fs.readFile);
```

Comment:
All of this about promises could be skipped if you use `fs.readFileSync`.

Reply:
CLI team frowns upon using sync IO operations. This is based on feedback I got from them for a command I developed a while back.